lib/wab/client.rb

```diff
@@ -0,0 +1,146 @@
+
+require 'net/http'
+require 'oj'
+
+module WAB
+
+  # A client for a WAB server. It is not specific to any particular
+  # runner. The client allows direct access to the server data. It is just
+  # another view implementation.
+  class Client
+    # Address of the WAB server.
+    attr_accessor :server_address
+    # The port the WAB serer is listening on.
+    attr_accessor :server_port
+    # Prefix to add to the URL path.
+    attr_accessor :path_prefix
+    # The URL path to execute TQL,
+    attr_accessor :tql_path
+    # The key for the type. The default is 'kind'.
+    attr_accessor :type_key
+    # If true the connection to the server is Keep-Alive.
+    attr_accessor :keep_alive
+
+    # Create a new Client for the server at +addr+ and +port+. The provided
+    # options should match the attribute names and types.
+    def initialize(addr, port, options={})
+      @server_address = addr
+      @server_port = port
+      @path_prefix = options.fetch(:path_prefix, '/v1/')
+      @tql_path = options.fetch(:tql_path, '/tql')
+      @type_key = options.fetch(:type_key, 'kind')
+      @keep_alive = options.fetch(:keep_alive, true)
```

Comment:
Some developers do not like using double negation `!!` but wouldn't this be better if it was..: ?
```ruby
@keep_alive = !!options.fetch(:keep_alive, true)
```